Improve how gender and birth year is displayed when changing the facility

MisRob opened the pull request:
## Summary
 
Utilize existing components for displaying gender and birth year in the _"Confirm account details"_ step when merging accounts in the change facility.

| Before | After |
| --------- | ------ |
| ![before](https://github.com/learningequality/kolibri/assets/13509191/0849720d-be8b-49f7-9917-49a5553e5692) | ![after](https://github.com/learningequality/kolibri/assets/13509191/cc50031e-d856-4c7c-9164-898c2f1931d9) |

## References

Resolves https://github.com/learningequality/kolibri/issues/10335

## Reviewer guidance

With _"On your own"_ device, go to your profile and change the facility by merging accounts (you can merge into _"test1"_ account with _"test1"_ password on _"Mr Tibbles' House of Phun"_ facility). Preview the _"Confirm account details"_ page.

![qa-steps](https://github.com/learningequality/kolibri/assets/13509191/3b0c5b9b-dbe1-412a-87b2-1a9955dc3d38)

---

## Testing checklist

- [x] Contributor has fully tested the PR manually
- [x] If there are any front-end changes, before/after screenshots are included
- [ ] Critical user journeys are covered by Gherkin stories
- [ ] Critical and brittle code paths are covered by unit tests


## PR process

- [x] PR has the correct target branch and milestone
- [x] PR has 'needs review' or 'work-in-progress' label
- [x] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

## Reviewer checklist

- Automated test coverage is satisfactory
- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md

MisRob committed:
Use existing components for  gender and birth year

Resolves https://github.com/learningequality/kolibri/issues/10335

rtibbles commented on kolibri/plugins/user_profile/assets/src/views/ChangeFacility/MergeAccountDialog/ConfirmAccountDetails.vue:
```diff
@@ -36,15 +36,15 @@
         </th>
         <td dir="auto" data-test="gender">
-          {{ cleanValue(targetAccount.gender) }}
+          <GenderDisplayText :gender="targetAccount.gender" />
```
Ah yes, much cleaner!

akolson replied:
➕

rtibbles reviewed (dismissed):
I had completely forgotten that these components existed, very simple, clean solution.

rtibbles commented:
To double check, I just went to look at the components in question, and it seems like the Gender component may have a bug:

https://github.com/learningequality/kolibri/blob/develop/kolibri/core/assets/src/views/userAccounts/GenderDisplayText.vue#L29 it is checking against `birthYear` here instead of `gender` for the second condition.

thanksameeelian commented:
i just wanted to comment that i love your use of a table here in your PR description, @MisRob! it makes the before/after comparison easier and is much nicer to look at, and even if it's a more common convention than i think, i haven't really seen a table used in this way in other PRs. after seeing this, i plan to start using tables for mine as well, when appropriate 🙂

akolson reviewed (commented):
Thanks @MisRob! Clean solution, and looks good to me! It's probably worth investing time to fix the bug in the gender component as spotted by Richard!

MisRob committed:
Fix condition

MisRob commented:
The condition is now fixed.

Also, I added another user to @rtibbles's facility you can try to merge your account into - username _"test2"_, password _"test2"_. This user has both birth year and gender specified.

MisRob commented:
Just a note that we will have QA on this, so let's wait before merging

rtibbles commented on kolibri/core/assets/src/views/userAccounts/GenderDisplayText.vue:
```diff
@@ -26,7 +26,7 @@
     computed: {
       isSpecified() {
-        return this.gender !== NOT_SPECIFIED && this.birthYear !== DEFERRED;
+        return this.gender !== NOT_SPECIFIED && this.gender !== DEFERRED;
```
:+1:

pcenov commented:
@radinamatic I confirm that this is implemented as specified in the description and didn't observe any related issues while manually testing.

radinamatic reviewed (approved):
Manual QA passes, good to go! 💯 👏🏽 :shipit:
